File: kotlin/kotlin-spring-boot/src/main/kotlin/com/meridal/examples/kotlinspringboot/controller/RecordingController.kt
package com.meridal.examples.kotlinspringboot.controller

import com.meridal.examples.kotlinspringboot.model.Recording
import com.meridal.examples.kotlinspringboot.service.RecordingService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*
import org.springframework.web.server.ResponseStatusException

@RestController
@RequestMapping("/api/recording")
class RecordingController @Autowired constructor(private val service: RecordingService) {

    @GetMapping("/")
    fun getAllRecordings() = service.getAllRecordings()

    @GetMapping("/{id}")
    fun getRecording(@PathVariable id: Long) = service.getRecording(id) ?:
        throw ResponseStatusException(HttpStatus.NOT_FOUND, "This recording does not exist")

    @PostMapping("/")
    fun saveRecording(@RequestBody recording: Recording): Recording = service.saveRecording(recording)

    @PutMapping("/{id}")
    fun updateRecording(@RequestBody recording: Recording): Recording = service.saveRecording(recording)

    @DeleteMapping("/{id}")
    fun deleteRecording(@PathVariable id: Long) = service.deleteRecording(id) ?:
        throw ResponseStatusException(HttpStatus.NOT_FOUND, "This recording does not exist")
}

File: scala/scala-spring-boot/src/main/scala/com/meridal/examples/scalaspringboot/controller/RecordingController.scala
package com.meridal.examples.scalaspringboot.controller

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation._
import org.springframework.web.server.ResponseStatusException
import org.springframework.http.HttpStatus

import scala.collection.JavaConverters._

import com.meridal.examples.scalaspringboot.model.Recording
import com.meridal.examples.scalaspringboot.service.RecordingService


@RestController
@RequestMapping(Array("/api/recording"))
class RecordingController(@Autowired val service: RecordingService) {

  @GetMapping(Array("/"))
  def getAllRecordings() = service.getAllRecordings().asJava

  @GetMapping(Array("/{id}"))
  def getRecording(@PathVariable id: Long): Recording = { 
    val recording = service.getRecording(id)
    if (recording == null) {
      throw new ResponseStatusException(HttpStatus.NOT_FOUND, "This recording does not exist")
    }
    return recording
  }

  @PostMapping(Array("/"))
  def saveRecording(@RequestBody recording: Recording) = service.saveRecording(recording)

  @PutMapping(Array("/{id}"))
  def updateRecording(@PathVariable id: Long, @RequestBody recording: Recording): Recording = {
    recording.id = id
    service.saveRecording(recording)
  }

  @DeleteMapping(Array("/{id}"))
  def deleteRecording(@PathVariable id: Long): Recording = {
    val recording = service.deleteRecording(id)
    if (recording == null) {
      throw new ResponseStatusException(HttpStatus.NOT_FOUND, "This recording does not exist") 
    }
    return recording
  } 
}
